### PR TITLE
Feature/pagination

### PR DESCRIPTION
**Pull request by nottoday2471**

### CODE REVIEW

File: src/components/pagination/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { FC } from 'react';
+import { Box, Pagination as PaginationMUI } from '@mui/material';
+
+type PaginationProp = {
```
Comment (BillyAleks) [inline src/components/pagination/Pagination.tsx:4]: Props is plural

File: src/components/pagination/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  totalCount: number;
+  getPage: (page: number) => void;
+};
+
+const Pagination: FC<PaginationProp> = (props: PaginationProp) => {
```
Comment (BillyAleks) [inline src/components/pagination/Pagination.tsx:9]:
```suggestion
const Pagination: FC<PaginationProp> = (props) => {
```

Comment (BillyAleks) [reply src/components/pagination/Pagination.tsx:9]: Why don't you extract it directly in a parameters? extract totalCount and getPage to reduce code amount later

File: src/pages/orders/Orders.tsx [hunk 3]
```diff
@@ -2,20 +2,38 @@ import { useEffect, useState } from 'react';
 import orderService from '../../api/orders.service';
 import { OrderCard } from '../../components/orders/OrderCard';
 import { IOrder } from '../../types/order.types';
-import style from './Orders.module.scss'
-
+import style from './Orders.module.scss';
+import Pagination from '../../components/pagination/Pagination';
```
Comment (BillyAleks) [inline src/pages/orders/Orders.tsx:6]: Just my recommendation of import order (not everyone use it, but looks more elegant):

1. Lib imports: "react"
2. Global folder imports: "Pages/components/my"
3. Sequence of imports from far neighbors to close neighbors:
"../../../../../test/utils";
"../../styles.ts";
"./helpers.ts";

File: src/pages/orders/Orders.tsx [hunk 4]
```diff
@@ -2,20 +2,38 @@ import { useEffect, useState } from 'react';
...
 
 export const Orders: React.FC = () => {
-    const [orders, setOrders] = useState([])
+  const [orders, setOrders] = useState([]);
```
Comment (BillyAleks) [inline src/pages/orders/Orders.tsx:9]: What kind of array is a type of orders?

File: src/pages/orders/Orders.tsx [hunk 5]
```diff
@@ -2,20 +2,38 @@ import { useEffect, useState } from 'react';
...
+    getOrders(page);
+  };
 
-    const getOrders = async () => {
-        await orderService.getOrders().then((data: any) => setOrders(data.data))
-    }
+  const getOrders = async (page = 1) => {
+    await orderService
+      .getOrders(undefined, page, undefined)
+      .then((data: any) => {
```
Comment (BillyAleks) [inline src/pages/orders/Orders.tsx:19]: Are you sure it's a right place to use any in here?

File: src/pages/orders/Orders.tsx [hunk 6]
```diff
@@ -2,20 +2,38 @@ import { useEffect, useState } from 'react';
...
+        setTotalPages(data.totalPages);
+        return data;
+      })
+      .then((data: any) => setOrders(data.data));
```
Comment (BillyAleks) [inline src/pages/orders/Orders.tsx:23]: same

File: src/api/orders.service.ts [hunk 7]
```diff
@@ -1,9 +1,9 @@
 import { axiosInstance } from './axios-instance';
 
 class OrderService {
-  async getOrders(limit = 10, page = 1) {
+  async getOrders(limit = 10, page = 1, sort = 'asc') {
     return await axiosInstance
```
Comment (BillyAleks) [inline src/api/orders.service.ts:5]: I think, there is no need to await for result in here. As you are chaining your Promise with a catch anyways

File: src/components/orders/OrderCard/OrderCard.tsx [hunk 8]
```diff
@@ -1,26 +1,28 @@
...
+import style from './OrderCard.module.scss';
 
 interface IOrderCardProps {
-    order: IOrder
+  order: IOrder;
 }
 
 export const OrderCard: React.FC<IOrderCardProps> = ({ order }) => {
-
   return (
```
Comment (BillyAleks) [inline src/components/orders/OrderCard/OrderCard.tsx:11]: return directly JSX

File: src/components/pagination/Pagination.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import { FC } from 'react';
+import { Box, Pagination as PaginationMUI } from '@mui/material';
+
+type PaginationProps = {
+  totalCount: number;
+  getPage: (page: number) => void;
+};
+
+const Pagination: FC<PaginationProps> = ({ totalCount, getPage }) => {
+  return (
```
Comment (BillyAleks) [inline src/components/pagination/Pagination.tsx:10]: Same here, you can return directly JSX

File: src/pages/orders/Orders.tsx [hunk 10]
```diff
@@ -1,21 +1,39 @@
...
-    await orderService.getOrders().then((data: any) => setOrders(data.data))
-  }
+  const getOrders = async (page = 1) => {
+    await orderService
+      .getOrders(undefined, page, undefined)
+      .then((data: IOrderResponse) => {
+        setTotalPages(data.totalPages);
+        return data;
+      })
+      .then((data: IOrderResponse) => setOrders(data.data));
```
Comment (BillyAleks) [inline src/pages/orders/Orders.tsx:23]: why didn't we do it in a previous then?

**Review by BillyAleks — COMMENTED** [review]
Thanks, please check the comments

**Review by RamboUA — APPROVED** [review]

**Pull request merged**